ovs/datapath.go

```diff
@@ -0,0 +1,338 @@
+// Copyright 2021 DigitalOcean.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ovs
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	errMissingDataPathName  = errors.New("datapath name argument is mandatory")
+	errUninitializedClient  = errors.New("client unitialized")
+	errMissingMandatoryZone = errors.New("at least 1 zone is mandatory")
+	errWrongArgumentNumber  = errors.New("missing or too much arguments to setup ct limits")
+	errWrongDefaultArgument = errors.New("wrong argument while setting default ct limits")
+	errWrongZoneArgument    = errors.New("wrong argument while setting zone ct limits")
+)
+
+// Zone defines the type used to store a zone as it is returned
+// by ovs-dpctl ct-*-limits commands
+type Zone map[string]uint64
+
+// ConnTrackOutput is a type defined to store the output
+// of ovs-dpctl ct-*-limits commands. For example it stores
+// such a cli output:
+// # ovs-dpctl ct-get-limits system@ovs-system zone=2,3
+// default limit=0
+// zone=2,limit=0,count=0
+// zone=3,limit=0,count=0
+type ConnTrackOutput struct {
+	// defaulCT is used to store the global setting: default
+	defaultLimit Zone
+	//zones stores all remaning zone's settings
+	zones []Zone
+}
+
+// DataPathReader is the interface defining the read operations
+// for the ovs DataPaths
+type DataPathReader interface {
+	// Version is the method used to get the version of ovs-dpctl
+	Version() (string, error)
+	// GetDataPath is the method that returns all DataPaths setup
+	// for an ovs switch
+	GetDataPath() ([]string, error)
+}
+
+// DataPathWriter is the interface defining the wrtie operations
+// for the ovs DataPaths
+type DataPathWriter interface {
+	// AddDataPath is the method used to add a datapath to the switch
+	AddDataPath(string) error
+	// DelDataPath is the method used to remove a datapath from the switch
+	DelDataPath(string) error
+}
+
+// ConnTrackReader is the interface defining the read operations
+// of ovs conntrack
+type ConnTrackReader interface {
+	// GetCTLimits is the method used to querying conntrack limits for a
+	// datapath on a switch
+	GetCTLimits(string, []uint64) (ConnTrackOutput, error)
+}
+
+// ConnTrackWriter is the interface defining the write operations
+// of ovs conntrack
+type ConnTrackWriter interface {
+	// SetCTLimits is the method used to setup a limit for an ofport (zone)
```

Comment:
I guess here you can specify only zone, since the per conntrack limit is not tied to the ofport